docs/tutorials/clojure-guide.md

```diff
@@ -0,0 +1,225 @@
+
+This is a feature recently introduced in LSP 3.16, which servers should return the call hierarchy of a function/method.
+As of now, `clojure-lsp` supports only incoming call hierarchy:
+
+![incoming call-hierarchy](images/clojure-call-hierarchy.png "incoming call-hierarchy")
+
+It uses [lsp-treemacs](https://github.com/emacs-lsp/lsp-treemacs) which you should install to use this feature.
+
+### Formatting
+
+For LSP formatting, `clojure-lsp` use [cljfmt](https://github.com/weavejester/cljfmt) with support for all known `cljfmt` 
+settings.
+
+![formatting](images/clojure-formatting.gif "formatting")
+
+### Code actions
+
+One of the most important features considered by me is the **code actions** feature provided by the server. they are refactoring actions
+that may change your code after applied, `clojure-lsp` some of the refactorings from below are suggested by the server depending
+where user cursor is, for example `Add missing namespace` where `clojure-lsp` will suggest adding a missing require:
+  
+![add-missing-ns](images/clojure-add-missing-ns.gif "add-missing-ns")
+
+### Refactorings
+
+`clojure-lsp` also has a lot of refactorings, some of them can be automatically triggered by 
+code actions like said above and others manually by user, one example of one of the most 
+used by me is `clean-ns`, which will remove any unused requires and refers:
+
+![clean-ns](images/clojure-clean-ns.gif "clean-ns")
+
+Or `rename` which will rename the symbol at point and all usages/references:
+
+![rename](images/clojure-rename.gif "rename")
+
+For more info on all available refactorings, check [here](https://github.com/clojure-lsp/clojure-lsp#refactorings).
+
+### Semantic tokens (Experimental)
+
+Another feature recently introduced in LSP 3.16, is **semantic tokens**, which consists of the LSP server return to 
+client the tokens and what they mean, with that, `lsp-mode` can color the tokens according to their meaning on the code:
+
+![semantic-tokens](images/clojure-semantic-tokens.png "semantic-tokens")
+
+## Performance
+
+For the first time `clojure-lsp` starts in your project, it will scan all jar dependencies and source code, this can 
+take a while depending on your project size, after started, it should be really fast and for the next time, it will use its cache and you may not notice any performance issues.
+
+## Server settings
+
+`clojure-lsp` has a lot of settings for improving your development experience, make sure to check it out the [settings](https://github.com/clojure-lsp/clojure-lsp#settings) section.
+For an example configuration, check [here](https://github.com/ericdallo/dotfiles/blob/master/.lsp/config.edn).
+
+## CIDER integration
+
+Therefore, it turns out that it's possible to use both `cider` and `lsp-mode` features for a clojure-development, both work 
+great together, I rely on `cider` when I really need a REPL, to eval anything, run tests, etc. You can just <kbd>M-x</kbd> `cider-jack-in` and the REPL will start normally.
```

Comment:
``` suggestion
We suggest using both `cider` and `lsp-mode` for Clojure development. 
I rely on `cider` to use its REPL, evaluate expressions, run tests, and more. You can execute <kbd>M-x</kbd> `cider-jack-in` to start the REPL.
```

Reply:
How about 'We suggest using `cider` and `lsp-mode` together for Clojure development.' and 'I rely on `cider` *for* its REPL, to evaluate expressions, to run tests and more'?